ansible/playbooks/roles/opendistro_for_elasticsearch/tasks/patch-log4j.yml

```diff
@@ -0,0 +1,68 @@
+---
+- name: Log4j patch
+  block:
+    - name: "opendistro_for_elasticsearch : Log4j patch | Get archive"
+      include_role:
+        name: download
+        tasks_from: download_file
+      vars:
+        file_name: "{{ log4j_file_name }}"
+
+    - name: Log4j patch | Extract archive
+      unarchive:
+        dest: /tmp/
+        src: "{{ download_directory }}/{{ log4j_file_name }}"
+        remote_src: true
+        list_files: true
+      register: unarchive_list_files
+
+    - name: Log4j patch | Copy new jars
+      register: log4j_patch
+      copy:
+        src: "{{ item.src }}"
+        dest: "{{ item.dest }}"
+        owner: elasticsearch
+        group: root
+        mode: u=rw,g=r,o=
+        remote_src: true
+      loop:
+        - { src: "{{ download_directory }}/{{ log4j_api }}",       dest: /usr/share/elasticsearch/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_api }}",       dest: /usr/share/elasticsearch/performance-analyzer-rca/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_api }}",       dest: /usr/share/elasticsearch/plugins/opendistro-performance-analyzer/performance-analyzer-rca/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_core }}",      dest: /usr/share/elasticsearch/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_core }}",      dest: /usr/share/elasticsearch/performance-analyzer-rca/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_core }}",      dest: /usr/share/elasticsearch/plugins/opendistro-performance-analyzer/performance-analyzer-rca/lib/ }
+        - { src: "{{ download_directory }}/{{ log4j_slfj_impl }}", dest: /usr/share/elasticsearch/plugins/opendistro_security/ }
+      vars:
+        log4j_core: "{{ unarchive_list_files.files | select('contains', 'log4j-api-2.17.0.jar') | first }}"
+        log4j_api: "{{ unarchive_list_files.files | select('contains', 'log4j-core-2.17.0.jar') | first }}"
+        log4j_slfj_impl: "{{ unarchive_list_files.files | select('contains', 'log4j-slf4j-impl-2.17.0.jar') | first }}"
+
+    - name: Log4j patch - cleanup
+      block:
+      - name: Log4j patch | Remove old jars
+        file:
+          state: absent
+          path: "{{ item }}"
+        loop:
+          - /usr/share/elasticsearch/plugins/opendistro-performance-analyzer/performance-analyzer-rca/lib/log4j-api-2.13.0.jar
+          - /usr/share/elasticsearch/plugins/opendistro-performance-analyzer/performance-analyzer-rca/lib/log4j-core-2.13.0.jar
+          - /usr/share/elasticsearch/performance-analyzer-rca/lib/log4j-api-2.13.0.jar
+          - /usr/share/elasticsearch/performance-analyzer-rca/lib/log4j-core-2.13.0.jar
+          - /usr/share/elasticsearch/lib/log4j-api-2.11.1.jar
+          - /usr/share/elasticsearch/lib/log4j-core-2.11.1.jar
+          - /usr/share/elasticsearch/plugins/opendistro_security/log4j-slf4j-impl-2.11.1.jar
+
```

Comment:
@rafzei are you sure if all the previous versions of epicli we currently support used exactly the same log4j versions? Wouldn't it be better to use some pattern?

Reply:
There were no changes between epicli (still supported) versions. The jars should be the same.